src/lib.rs

```diff
@@ -170,7 +170,21 @@ impl AutoCfg {
         // so for now we only apply RUSTFLAGS when cross-compiling an artifact.
         //
         // See https://github.com/cuviper/autocfg/pull/10#issuecomment-527575030.
-        let rustflags = if env::var_os("TARGET") != env::var_os("HOST") {
+        let dir_contains_target = dir
+            .to_str()
+            .and_then(|dir| {
+                env::var("TARGET").ok().map(|target| {
+                    let mut cargo_target_dir =
+                        env::var("CARGO_TARGET_DIR").unwrap_or_else(|_| "target".to_string());
+                    cargo_target_dir.push(MAIN_SEPARATOR);
+                    cargo_target_dir.push_str(&target);
+
+                    dir.contains(&cargo_target_dir)
```

Comment:
Too bad that neither `Path` nor `OsStr` have a `contains` method, but I guess UTF-8 will almost always be fine anyway. `Path::starts_with` would be ideal, but we'd have to `canonicalize` first, which isn't available until Rust 1.5.
(Curse my ambition to 1.0 compatibility...)

Reply:
> Path::starts_with would be ideal

I used that in my first try but it does not work as `dir` is an absolute path and hence does usually not start with `$CARGO_TARGET_DIR/$TARGET` does only contain it. This is why I switch to the `str::contains` after dabbling a bit with manually implementing it on the iterators returned by `Path::components`.